### PR TITLE
use next image instead of <img> for image optimization; fix mobile view of SDA post

### PR DESCRIPTION
**Pull request by yuhan**
before:
image

after:
image

### CODE REVIEW

File: components/CustomMdxProvider.tsx [hunk 1]
```diff
@@ -22,6 +23,9 @@ const mdxComponents: MDXProviderComponentsProp = {
   h4: withScrollMargin('h4'),
   h5: withScrollMargin('h5'),
   h6: withScrollMargin('h6'),
+  img: ({ children, ...props }) => (
+    <Image {...props} layout="responsive" loading="lazy" unsized={true} />
```
Comment (yuhan) [inline components/CustomMdxProvider.tsx:27]: cc @smackesey 
we can move the https://github.com/dagster-io/dagster-website/pull/115

also, FYI, once you move stuff here, the dev console is likely to throw some validateDOMNesting warnings. so be sure to address these warnings.
see details here: https://github.com/dagster-io/dagster/pull/6722#discussion_r810327596

File: pages/blog/software-defined-assets.mdx [hunk 2]
```diff
@@ -184,7 +184,7 @@ An asset-based orchestrator addresses both of these gaps by combining polyglot c
 
 For a video version of this section, watch:
 
-<iframe width="560" height="315" src="https://www.youtube.com/embed/eS--8brw5YM" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+<iframe width="100%" height="315" src="https://www.youtube.com/embed/eS--8brw5YM" title="YouTube video player" frameBorder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowFullScreen></iframe>
```
Comment (yuhan) [inline pages/blog/software-defined-assets.mdx:187]: this fixes the mobile view of the SDA post. mobile would be off when the screen width is smaller than the fixed width 560.
cc @sryza 


before:
image


after: https://dagster-website-3z0ztvl3r-elementl.vercel.app/blog/software-defined-assets#loading-asset-definitions-from-modern-data-stack-tools
image

**Comment by braunjj** [comment]
Looks good to me but I am not able to get the video to play when I tap on it from my phone for some reason. It works if i view it in landscape mode though, maybe the issue is with the responsive layout around the video?

**Comment by braunjj** [comment]
> Looks good to me but I am not able to get the video to play when I tap on it from my phone for some reason. It works if i view it in landscape mode though, maybe the issue is with the responsive layout around the video?
> 

 Looks like this is working fine now

**Review by braunjj — APPROVED** [review]

**Pull request merged**